executor/cte.go

```diff
@@ -657,3 +655,29 @@ func (p *cteProducer) checkHasDup(probeKey uint64,
 	}
 	return false, nil
 }
+
+// Return true if cor col has changed.
+func (p *cteProducer) checkAndUpdateCorColHashCode() bool {
+	var changed bool
+	for i, corCol := range p.corCols {
+		corCol.CleanHashCode()
+		newHashCode := corCol.HashCode(p.ctx.GetSessionVars().StmtCtx)
+		if !sameHashCode(newHashCode, p.corColHashCodes[i]) {
+			changed = true
+			p.corColHashCodes[i] = newHashCode
+		}
+	}
+	return changed
+}
+
+func sameHashCode(c1 []byte, c2 []byte) bool {
+	if len(c1) != len(c2) {
+		return false
+	}
+	for i := 0; i < len(c1); i++ {
+		if c1[i] != c2[i] {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
Why do we need this instead of using `bytes.Compare()`?

Reply:
Done